database/src/main/dataset_schema/add_schema.sql

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP FUNCTION IF EXISTS dataset_schema.add_schema(TEXT, INTEGER, TEXT, JSONB, TEXT);
+
+CREATE OR REPLACE FUNCTION dataset_schema.add_schema(
+    IN i_schema_name        TEXT,
+    IN i_schema_version     INTEGER,
```

Comment:
So the version is always `latest + 1`. Thinking if a better info isn't `current latest`. Need to get back to this

Reply:
But! I think this approach is better for two reasons.
If incrementing would be done within DB, it would be nice to return the new version(of course still possible, or maybe not necessary).
The function name would be weird for new schemas.